core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -291,6 +292,20 @@ public <T extends ConnectionDetails> boolean test( T connectionDetails ) throws
     return connectionProvider.test( connectionDetails );
   }
 
+  /**
+   * Run a test operation on the named connection
+   *
+   * @param connectionDetails The named connection details to test
+   * @return A boolean signifying the success of the test operation
+   */
+  @SuppressWarnings( "unchecked" )
+  public <T extends VFSConnectionDetails> boolean test(T connectionDetails, VFSConnectionTestOptions options )
```

Comment:
`ConnectionManager#test` is public function need to handle all null arguments 

-` connectionDetails`
-` options`

Reply:
addressed the changes.

Reply:
@NJtwentyone I'm not sure what was your intention here.

I think it is reasonable to get an NPE when `connectionDetails` is passed null. It should be marked `@NonNull`. I usually use `Objects.requireNonNull()` right at the top, but the only thing it achieves is to antecipate the NPE, and used for cases where a normal NPE would only occur later, elsewhere. In this method, if given null, the exception would occur immediately, when dereferencing `connectionDetails.getType()`. You can argue, though, that we should instead be throwing an `IllegalArgumentException` instead.

Regarding the `options` argument, I think it actually makes sense for it to be `@Nullable` (and be marked with an annotation). When `null`, we would create an instance of `VFSConnectionTestOptions` (accepting its default state). Then, we could even change the `VFSConnectionProvider#test(VFSConnectionDetails, VFSConnectionTestOptions)` to be `@NonNull` for both arguments.

Reply:
Looks addressed to me.

Reply:
addressed in https://github.com/pentaho/pentaho-kettle/commit/706cfc42fcd1555d6b83faceecfb3047f3f4ecf7